### ISSUE TITLE
Pylint Incorrectly Marks Required[...] and NotRequired[...] TypedDict types as "Undefined-Variable"; PEP 655

### ISSUE THREAD

**Issue by mjohnson159** [PyCQA/pylint#8048]
### Bug description

[PEP 655 reference](https://peps.python.org/pep-0655/) and [typing.Required, typing.NotRequired doc reference](https://docs.python.org/3.11/library/typing.html?highlight=static%20typing#typing.Required)

When creating a `TypedDict` in python 3.11+, PEP 655 allows specifying of `NotRequired[]` types for `total=True` and `Required[]` types when `total=False`. Currently, these are flagged as "undefined-variable" by Pylint when they should register as other types, e.g. `(class) list`, `(class) str`, etc.

Examples

```py
from typing import TypedDict

class TESTCLASS(TypedDict):
    val1: NotRequired[str]
    val2: str  # required variable

class TESTCLASS2(TypedDict, total=False):
    val1: Required[str]
    val2: str  # unrequired variable
```
I searched multiple different ways and keywords to find an issue with this, via both google and github issues search bar, but I was unable to find anything pertaining to this. Apologies if there's already an opened issue for this.

### Command used

```shell
pylint test.py
```

### Pylint output

```shell
************* Module test
test.py:4:10: E0602: Undefined variable 'NotRequired' (undefined-variable)
test.py:8:10: E0602: Undefined variable 'Required' (undefined-variable)
```

### Expected behavior

The values would be allowed to define valid Types for TypedDict classes.

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.12.13
Python 3.11.1 (main, Dec 23 2022, 09:40:27) [Clang 14.0.0 (clang-1400.0.29.202)]
```

### OS / Environment

MacOS 12.6.2 Monterey
zsh 5.8.1 (x86_64-apple-darwin21.0)

**Comment by jacobtylerwalls** [comment]
Don't they have to be imported?

```python
$ python3.11 a.py
Traceback (most recent call last):
  File "/Users/user/a.py", line 3, in <module>
    class TESTCLASS(TypedDict):
  File "/Users/user/a.py", line 4, in TESTCLASS
    val1: NotRequired[str]
          ^^^^^^^^^^^
NameError: name 'NotRequired' is not defined
```

**Comment by mbyrnepr2** [comment]
Thanks for the report @mjohnson159  but I think we can close per @jacobtylerwalls's comment - `undefined-variable` does not occur if the types are imported with:
```python
from typing import Required, NotRequired
```

**Comment by mjohnson159** [comment]
Strangely, I had tried importing them previously to no avail. I'm not sure what I was doing, but I tried triple checking everything before opening this. Apologies for wasting your time.
